### DISCUSSION TITLE
Interrupted pruning/compacting

### DISCUSSION THREAD

**Discussion by sophie-h** [borgbackup/borg#6283] [Q&A] [answered]
It does look like borg prune has to start all over again if it is canceled while running in 1.1. Is that correct?

Will borg compact in borg 1.2 be able to continue where it was left off?

I'm currently trying to prune hundreds of archives but there does not seem to be an easy way to complete this.

**Comment by ThomasWaldmann** [comment, marked as answer]
if you interrupt it before it commits, yes, all will start from the last valid commit before that (IIRC).

there are 2 commits (borg < 1.2) or 1 commit (borg >= 1.2):
- after logically deleting the archives (all borg versions)
- after `compact_segments()` (only in borg < 1.2). in borg >= 1.2, this is separated into a `borg compact` run.

**Reply by sophie-h** [reply]
Thanks for the clarification. How difficult would it be to add something akin to `--checkpoint-interval` for regular commits during `prune` and `compact`?

**Reply by ThomasWaldmann** [reply]
didn't look yet, but let's continue in #6284.

**Comment by ThomasWaldmann** [comment]
the easiest way to avoid having to delete a huge amount of archives is of course running `borg prune` regularly.
